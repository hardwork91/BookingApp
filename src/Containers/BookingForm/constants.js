export const TOOLTIP_DELAY = 1;
export const ORIGIN_SELECTED_DATA = "originSelectedData";
export const DESTINATION_SELECTED_DATA = "destinationSelectedData";

export const ROUND_TRIP = "Round-trip";
export const ONE_WAY = "One-way";
export const MULTI_CITY = "Multi-city";
export const TRIP_BUILDER = "Trip Builder";

export const FLIGHT_TYPES = [ROUND_TRIP, ONE_WAY, MULTI_CITY, TRIP_BUILDER];

export const ECONOMY = "Economy";
export const BUSINESS = "Business";
export const FIRST = "First";

export const TRAVEL_CLASES = [ECONOMY, BUSINESS, FIRST];
