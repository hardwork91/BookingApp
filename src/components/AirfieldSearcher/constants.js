export const MAX_AIRPORT_ALLOWED = 3;

export const SEARCH_AIRFIELD_REQUEST = "SEARCH_AIRFIELD_REQUEST";
export const SEARCH_AIRFIELD_SUCCESS = "SEARCH_AIRFIELD_SUCCESS";
export const SEARCH_AIRFIELD_FAILURE = "SEARCH_AIRFIELD_FAILURE";

export const CLEAR_AIRFIELD_REQUEST = "CLEAR_AIRFIELD_REQUEST";
