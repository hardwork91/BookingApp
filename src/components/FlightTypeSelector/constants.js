export const ROUND_TRIP = "Round-trip";
export const ONE_WAY = "One-way";
export const MULTI_CITY = "Multi-city";
export const TRIP_BUILDER = "Trip Builder";

export const FLIGHT_TYPES = [ROUND_TRIP, ONE_WAY, MULTI_CITY, TRIP_BUILDER];
