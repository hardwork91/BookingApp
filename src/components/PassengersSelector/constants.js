// All categories are exported individually in case of being needed
// Each gategory has Name(used as label and key) and Nescription
const CHILD = "Child";
export const CHILD_DESCRIPTION = "- 11";
const YOUTH = "Youth";
export const YOUTH_DESCRIPTION = "12 - 17";
const ADULTS = "Adults";
export const ADULTS_DESCRIPTION = "18 - 64";
const SENIORS = "Seniors";
export const SENIORS_DESCRIPTION = "65 +";

// Add a new category is easy, just need to add it on the PASSENGERS_CATEGORIES array
export const PASSENGERS_CATEGORIES = [CHILD, YOUTH, ADULTS, SENIORS];

// Itinal values also easy to modify
export const PASSENGERS_INITIAL_VALUE = {
  [CHILD]: 0,
  [YOUTH]: 0,
  [ADULTS]: 1,
  [SENIORS]: 0,
};
