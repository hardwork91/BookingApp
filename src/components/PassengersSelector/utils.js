import React from "react";
import { Typography, Popover, Icon } from "antd";

const { Text } = Typography;

/* displayed label of passengers selector control changes acording the selected categories.
 * this functions iterates over the current selected passengers and defines which is the propertly label to use.
 * gramatical sintaxis could be better, but we decide to keep it as simple as possible.
 */
export const generatePassengerSelectorLabel = passengers => {
  let categoriesWithAmount = 0;
  let totalAmount = 0;
  let singleCategory;
  for (const category in passengers) {
    const amount = passengers[category];
    if (amount > 0) {
      // check if there are more than one category selected
      categoriesWithAmount++;
      singleCategory = category;
    }
    totalAmount += amount;
  }

  // For correct gramatical sintaxis shuld be propertly to use formatMessage form React-intl.And usig formatters like "plural".
  // In this exaple we are not using it in order to keep the aplication simple due to internacionalization is not mandatory.
  let label =
    categoriesWithAmount > 1
      ? `${totalAmount} Travelers`
      : `${totalAmount} ${singleCategory}`;

  // if there is any category selected, the font color is marked in red. Alsoa a warning message is displayed over the componet to help color blind users.
  if (categoriesWithAmount === 0) {
    label = (
      <Popover
        content={
          <span>
            <Icon
              type="warning"
              style={{ color: "#f5222d", marginRight: "10px" }}
            />
            You must select at least one passenger
          </span>
        }
        trigger="click"
        defaultVisible
      >
        <Text type="danger">{`${totalAmount} Travelers`}</Text>
      </Popover>
    );
  }

  return {
    label,
    amount: totalAmount,
  };
};
