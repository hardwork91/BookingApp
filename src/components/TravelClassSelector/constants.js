export const ECONOMY = "Economy";
export const BUSINESS = "Business";
export const FIRST = "First";

export const TRAVEL_CLASES = [ECONOMY, BUSINESS, FIRST];
