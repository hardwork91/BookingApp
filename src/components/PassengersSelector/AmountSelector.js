import React, { Component } from "react";
import PropTypes from "prop-types";
import { InputNumber, Typography } from "antd";
import * as constants from "./constants";

const { Text } = Typography;

export default class AmountSelector extends Component {
  // update the passenger selector state when change the spinner value
  handleChange = value => {
    const { category, onChange } = this.props;
    onChange(category, value);
  };

  render() {
    const { category, amount } = this.props;
    return (
      <div>
        <Text style={{ marginRight: ".5em" }}>{category}</Text>
        <Text type="secondary">
          {constants[`${category.toUpperCase()}_DESCRIPTION`]}
        </Text>
        <InputNumber
          style={{ float: "right", marginLeft: "1em" }}
          size="small"
          // imput number min value is setted to 0
          min={0}
          defaultValue={amount}
          onChange={this.handleChange}
          precision={0}
        />
      </div>
    );
  }
}

// prop types validation
AmountSelector.propTypes = {
  category: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  amount: PropTypes.number.isRequired,
};
