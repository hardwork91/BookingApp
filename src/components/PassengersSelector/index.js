import React, { Component } from "react";
import PropTypes from "prop-types";
import { Icon, Menu, Dropdown, Tooltip } from "antd";
import AmountSelector from "./AmountSelector";
import { generatePassengerSelectorLabel } from "./utils";
import { PASSENGERS_CATEGORIES } from "./constants";
import { TOOLTIP_DELAY } from "../../Containers/BookingForm/constants";

export default class PassengersSelector extends Component {
  constructor(props) {
    super(props);
    this.state = {
      visible: false,
      passengers: props.value,
    };
  }

  // the visibility must be controlled manually to avoid getting the menu closed when the input number is clicked
  handleVisibleChange = flag => {
    // the visibility is stored in the component state
    this.setState({ visible: flag });
  };

  // when the amout of any category changes, it must be updated in the state and also update form item value.
  handleAmountChange = (category, amount) => {
    const { onChange, handleClearResults } = this.props;
    const passengers = { ...this.state.passengers, [category]: amount };
    this.setState(
      {
        passengers,
      },
      () => {
        // update the form field value
        onChange(passengers);
        // on value change, the booking results must be cleared
        handleClearResults();
      }
    );
  };

  // prevent default <a/> tag behavior
  handleOnClick = e => e.preventDefault();

  // render the menu using the PASSENGERS_CATEGORIES preset list of categories
  renderMenu = () => (
    <Menu>
      {PASSENGERS_CATEGORIES.map(category => {
        return (
          <Menu.Item key={category}>
            <AmountSelector
              category={category}
              amount={this.state.passengers[category]}
              onChange={this.handleAmountChange}
            />
          </Menu.Item>
        );
      })}
    </Menu>
  );

  render() {
    // The control label is generated depending on how many passengers and categories the user selects
    const { label } = generatePassengerSelectorLabel(this.state.passengers);

    return (
      // use a tooltip for helping users to understand what means this control
      <Tooltip title="Passengers" mouseEnterDelay={TOOLTIP_DELAY}>
        <Dropdown
          disabled={this.props.disabled}
          overlay={this.renderMenu()}
          trigger={["click"]}
          visible={this.state.visible}
          onVisibleChange={this.handleVisibleChange}
        >
          <a className="ant-dropdown-link" onClick={this.handleOnClick}>
            {label}
            <Icon type="down" />
          </a>
        </Dropdown>
      </Tooltip>
    );
  }
}

// prop types validation
PassengersSelector.propTypes = {
  value: PropTypes.object,
  onChange: PropTypes.func,
  handleClearResults: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};
