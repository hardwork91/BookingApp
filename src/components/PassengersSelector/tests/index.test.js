import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { mount, configure } from "enzyme";
import toJson from "enzyme-to-json";
import PassengersSelector from "../index";
import { PASSENGERS_INITIAL_VALUE, CHILD } from "../constants";

configure({ adapter: new Adapter() });

const setup = props => ({
  value: PASSENGERS_INITIAL_VALUE,
  onChange: jest.fn(),
  handleClearResults: jest.fn(),
  ...props,
});

describe("PassengersSelector", () => {
  it("render as expected", () => {
    const tree = mount(<PassengersSelector {...setup()} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("when handleVisibleChange", () => {
    const tree = mount(<PassengersSelector {...setup()} />);
    tree.instance().handleVisibleChange(true);
    expect(tree.state().visible).toBeTruthy();
  });

  it("when handleAmountChange", () => {
    const props = setup();
    const tree = mount(<PassengersSelector {...props} />);
    tree.instance().handleAmountChange(CHILD, 1);
    expect(props.onChange).toBeCalled();
    expect(props.handleClearResults).toBeCalled();
  });

  it("when handleOnClick", () => {
    const tree = mount(<PassengersSelector {...setup()} />);
    const preventDefault = jest.fn();
    tree.instance().handleOnClick({ preventDefault });
    expect(preventDefault).toBeCalled();
  });
});
