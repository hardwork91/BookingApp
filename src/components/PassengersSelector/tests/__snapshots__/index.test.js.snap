// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PassengersSelector render as expected 1`] = `
<PassengersSelector
  handleClearResults={[Function]}
  onChange={[Function]}
  value={
    Object {
      "Adults": 1,
      "Child": 0,
      "Seniors": 0,
      "Youth": 0,
    }
  }
>
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    mouseEnterDelay={1}
    mouseLeaveDelay={0.1}
    placement="top"
    title="Passengers"
    transitionName="zoom-big-fast"
  >
    <Tooltip
      align={Object {}}
      arrowContent={null}
      arrowPointAtCenter={false}
      autoAdjustOverflow={true}
      builtinPlacements={
        Object {
          "bottom": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tc",
              "bc",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "bottomLeft": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tl",
              "bl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "bottomRight": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tr",
              "br",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "left": Object {
            "ignoreShake": true,
            "offset": Array [
              -4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "cr",
              "cl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "leftBottom": Object {
            "ignoreShake": true,
            "offset": Array [
              -4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "br",
              "bl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "leftTop": Object {
            "ignoreShake": true,
            "offset": Array [
              -4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tr",
              "tl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "right": Object {
            "ignoreShake": true,
            "offset": Array [
              4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "cl",
              "cr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "rightBottom": Object {
            "ignoreShake": true,
            "offset": Array [
              4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bl",
              "br",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "rightTop": Object {
            "ignoreShake": true,
            "offset": Array [
              4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tl",
              "tr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "top": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              -4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bc",
              "tc",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "topLeft": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              -4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bl",
              "tl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "topRight": Object {
            "ignoreShake": true,
            "offset": Array [
              0,
              -4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "br",
              "tr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
        }
      }
      destroyTooltipOnHide={false}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.1}
      onPopupAlign={[Function]}
      onVisibleChange={[Function]}
      overlay="Passengers"
      placement="top"
      prefixCls="ant-tooltip"
      title="Passengers"
      transitionName="zoom-big-fast"
      trigger={
        Array [
          "hover",
        ]
      }
      visible={false}
    >
      <Trigger
        action={
          Array [
            "hover",
          ]
        }
        afterPopupVisibleChange={[Function]}
        arrowContent={null}
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        blurDelay={0.15}
        builtinPlacements={
          Object {
            "bottom": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tc",
                "bc",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "bottomLeft": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tl",
                "bl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "bottomRight": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tr",
                "br",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "left": Object {
              "ignoreShake": true,
              "offset": Array [
                -4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "cr",
                "cl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "leftBottom": Object {
              "ignoreShake": true,
              "offset": Array [
                -4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "br",
                "bl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "leftTop": Object {
              "ignoreShake": true,
              "offset": Array [
                -4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tr",
                "tl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "right": Object {
              "ignoreShake": true,
              "offset": Array [
                4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "cl",
                "cr",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "rightBottom": Object {
              "ignoreShake": true,
              "offset": Array [
                4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bl",
                "br",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "rightTop": Object {
              "ignoreShake": true,
              "offset": Array [
                4,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tl",
                "tr",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "top": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                -4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bc",
                "tc",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "topLeft": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                -4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bl",
                "tl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "topRight": Object {
              "ignoreShake": true,
              "offset": Array [
                0,
                -4,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "br",
                "tr",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
          }
        }
        defaultPopupVisible={false}
        destroyPopupOnHide={false}
        focusDelay={0}
        getDocument={[Function]}
        getPopupClassNameFromAlign={[Function]}
        hideAction={Array []}
        mask={false}
        maskClosable={true}
        mouseEnterDelay={1}
        mouseLeaveDelay={0.1}
        onPopupAlign={[Function]}
        onPopupVisibleChange={[Function]}
        overlay="Passengers"
        popup={[Function]}
        popupAlign={Object {}}
        popupClassName=""
        popupPlacement="top"
        popupStyle={Object {}}
        popupTransitionName="zoom-big-fast"
        popupVisible={false}
        prefixCls="ant-tooltip"
        showAction={Array []}
        title="Passengers"
        visible={false}
      >
        <Dropdown
          key="trigger"
          mouseEnterDelay={0.15}
          mouseLeaveDelay={0.1}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onVisibleChange={[Function]}
          overlay={
            <Menu>
              <MenuItem>
                <AmountSelector
                  amount={0}
                  category="Child"
                  onChange={[Function]}
                />
              </MenuItem>
              <MenuItem>
                <AmountSelector
                  amount={0}
                  category="Youth"
                  onChange={[Function]}
                />
              </MenuItem>
              <MenuItem>
                <AmountSelector
                  amount={1}
                  category="Adults"
                  onChange={[Function]}
                />
              </MenuItem>
              <MenuItem>
                <AmountSelector
                  amount={0}
                  category="Seniors"
                  onChange={[Function]}
                />
              </MenuItem>
            </Menu>
          }
          placement="bottomLeft"
          trigger={
            Array [
              "click",
            ]
          }
          visible={false}
        >
          <Dropdown
            defaultVisible={false}
            mouseEnterDelay={0.15}
            mouseLeaveDelay={0.1}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onVisibleChange={[Function]}
            overlay={[Function]}
            overlayClassName=""
            overlayStyle={Object {}}
            placement="bottomLeft"
            prefixCls="ant-dropdown"
            showAction={Array []}
            transitionName="slide-up"
            trigger={
              Array [
                "click",
              ]
            }
            visible={false}
          >
            <Trigger
              action={
                Array [
                  "click",
                ]
              }
              afterPopupVisibleChange={[Function]}
              blurDelay={0.15}
              builtinPlacements={
                Object {
                  "bottomCenter": Object {
                    "offset": Array [
                      0,
                      4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "tc",
                      "bc",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                  "bottomLeft": Object {
                    "offset": Array [
                      0,
                      4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "tl",
                      "bl",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                  "bottomRight": Object {
                    "offset": Array [
                      0,
                      4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "tr",
                      "br",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                  "topCenter": Object {
                    "offset": Array [
                      0,
                      -4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "bc",
                      "tc",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                  "topLeft": Object {
                    "offset": Array [
                      0,
                      -4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "bl",
                      "tl",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                  "topRight": Object {
                    "offset": Array [
                      0,
                      -4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "br",
                      "tr",
                    ],
                    "targetOffset": Array [
                      0,
                      0,
                    ],
                  },
                }
              }
              defaultPopupVisible={false}
              defaultVisible={false}
              destroyPopupOnHide={false}
              focusDelay={0}
              getDocument={[Function]}
              getPopupClassNameFromAlign={[Function]}
              hideAction={Array []}
              mask={false}
              maskClosable={true}
              mouseEnterDelay={0.15}
              mouseLeaveDelay={0.1}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onPopupAlign={[Function]}
              onPopupVisibleChange={[Function]}
              onVisibleChange={[Function]}
              overlay={[Function]}
              popup={[Function]}
              popupAlign={Object {}}
              popupClassName=""
              popupPlacement="bottomLeft"
              popupStyle={Object {}}
              popupTransitionName="slide-up"
              popupVisible={false}
              prefixCls="ant-dropdown"
              showAction={Array []}
              visible={false}
            >
              <a
                className="ant-dropdown-link ant-dropdown-trigger"
                key="trigger"
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
              >
                1 Adults
                <Icon
                  type="down"
                >
                  <LocaleReceiver
                    componentName="Icon"
                  >
                    <i
                      aria-label="icon: down"
                      className="anticon anticon-down"
                    >
                      <IconReact
                        className=""
                        type="down-o"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          key="svg-down"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            key="svg-down-svg-0"
                          />
                        </svg>
                      </IconReact>
                    </i>
                  </LocaleReceiver>
                </Icon>
              </a>
            </Trigger>
          </Dropdown>
        </Dropdown>
      </Trigger>
    </Tooltip>
  </Tooltip>
</PassengersSelector>
`;
